### ISSUE TITLE
False positive (unused key) with next-intl

### ISSUE THREAD

**Issue by arananegra** [lingualdev/i18n-check#83]
I've been trying this project and found some keys we weren't using.

I've detected some cases when the execution seems to find unused keys. However, those keys are being used. The problem seems to be related to using the key in a generateMetadata function in nextjs:

Example:

```
import { Metadata } from 'next';
import { getTranslations } from 'next-intl/server';

import { UserProfileComponent } from '../../../../pods/user-profile/user-profile.component';

export const generateMetadata = async ({
	params,
}: {
	params: Promise<{ username: string; locale: string }>;
}): Promise<Metadata> => {
	const { username, locale } = await params;
	const t = await getTranslations({ locale });
	return {
		title: t('{username} profile', { username }),
	};
};

export default async function UserProfilePage({ params }: { params: Promise<{ username: string }> }) {
	const { username } = await params;

	return <UserProfileComponent username={username} />;
}
```

If we then run i18n-check we get this: `i18n-check -l src/core/i18n/dictionaries --source es -f next-intl -u src`

```
No missing keys found!

No invalid translations found!

Found unused keys!
┌────────────────────────────────────┬───────────────────────────────┐
│ file                               │ key                           │
├────────────────────────────────────┼───────────────────────────────┤
│ src/core/i18n/dictionaries/es.json │ {username} profile            │
.................
└────────────────────────────────────┴───────────────────────────────┘

No undefined keys found!

Done in 0.31s.

```

It's clear that those keys are being used, but the library seems unable to detect them. Also, I've tried with a key without a variabl,e and the problem is the same (don't think it's related to that)

**Comment by busypeoples** [comment]
We will look into the issue.

**Comment by arananegra** [comment]
it's a function used from nextjs: https://nextjs.org/docs/app/api-reference/functions/generate-metadata

**Comment by busypeoples** [comment]
We will add your example as a test scenario and implement a fix for this.

**Comment by busypeoples** [comment]
@arananegra Alternatively we could also try to mark any interpolated keys as dynamic and skip them from any comparison.

**Comment by busypeoples** [comment]
@arananegra if you don't want to use the exclude option, you could try out using comments instead:

```ts
export const generateMetadata = async ({
	params,
}: {
	params: Promise<{ username: string; locale: string }>;
}): Promise<Metadata> => {
	const { username, locale } = await params;
	const t = await getTranslations({ locale });
        
        // t("SomeName profile")
        // t("SomeOtherName profile")
        // t("AnotherName profile")
	return {
		title: t('{username} profile', { username }),
	};
};
```

The parser can find any t function calls inside a single comment line, but this requires to know what the dynamic keys are (which we can assume they are known as there is a locale file with the specific keys).

Let us know if that works for you.

**Comment by busypeoples** [comment]
@arananegra if you can send me an example of how the key is defined in your locales files. Is the key itself also dynamic as in your example? If the setup like [example in the docs](https://next-intl.dev/docs/usage/translations#arguments), that would be easier - as the key is static, but the message is interpolated. That would work out of the box.
As mentioned, if you can provide a specific example, we can try to see what we could additionally do.
